state/factory/workingset.go

```diff
@@ -404,6 +404,40 @@ func (ws *workingSet) process(ctx context.Context, actions []action.SealedEnvelo
 	return ws.finalize()
 }
 
+func (ws *workingSet) validateSystemActionLayout(ctx context.Context, actions []action.SealedEnvelope) error {
+	reg := protocol.MustGetRegistry(ctx)
+	postSystemActions := []action.Envelope{}
+	for _, p := range reg.All() {
+		if psc, ok := p.(protocol.PostSystemActionsCreator); ok {
+			elps, err := psc.CreatePostSystemActions(ctx, ws)
+			if err != nil {
+				return err
+			}
+			postSystemActions = append(postSystemActions, elps...)
+		}
+	}
```

Comment:
unify the generation of system actions (not signed) in a private function

Reply:
> unify the generation of system actions (not signed) in a private function

OK